src/org/mozilla/javascript/optimizer/Block.java

```diff
@@ -529,6 +529,7 @@ private static int findExpressionType(OptFunctionNode fn, Node n,
             case Token.ARRAYCOMP:
             case Token.ARRAYLIT:
             case Token.OBJECTLIT:
+            case Token.BIGINT:
```

Comment:
I wasn't sure if `BigInt` should be `AnyType` or not.
Should these also be `AnyType` as the result can be `BigInt`?
https://github.com/mozilla/rhino/blob/21198a516ee818c56dbd689198fdfcb4ec3dc79e/src/org/mozilla/javascript/optimizer/Block.java#L482-L497